enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,316 @@
+
+Additionally, any error within the backup sidecar, the container goes into `CrashLoopBackOff`, which leads to the whole `Pod` reports `NotReady` status. Consequently, the whole `Etcd-cluster` can not server requests, nor the `API-Server`.
+
+
+#### DaemonSet manages an Etcd-backup pod within every control-plane node.
+
+This approach separates the `etcd-backup-server` deployment from the etcd static pods, to avoid affecting etcd-cluster and API-Server availability upon any error within the etcd-backup-server.
+
+The advantage of using `DaemonSet` over Deployment is to guarantee scheduling etcd-backup-server pods on all `Control-plane` nodes, regardless of the number of nodes (e.g. 1, 3, 5) using `nodeSelector` capability.
+By setting the `nodeSelector` to the master node label (i.e. `node-role.kubernetes.io/master`), etcd-backup-server's pods are guaranteed to scheduled on every control-plane node.
+
+This method has been implemented and tested. The same container used in the Sidecar approach has been utilised. However, the scheduling is being managed by a DaemonSet.
+Upon creating a `Backup` CR on an Openshift cluster, the `PeriodicBackupController` reacts by creating a DaemonSet resource, with the `etcd-backup-server` container and the configurations applied in the CR.
+During [testing](https://github.com/openshift/cluster-etcd-operator/pull/1354#issuecomment-2408661261), this method has shown a solid and stable deployment, where the Backup pods shown only `1` restart over a duration of `45` minutes. Moreover, the cause of the restart on all the Backup pods were due to `node` issue, not the backup container.
+
+_**Pros**_
+
+As the Backup pods are being scheduled independently of the etcd pods, any error within the Backup pods does not influence cluster Openshift cluster availability.
+
+Moreover, as the Backup pods are being scheduled within every master nodes, the backups are being taken from every local etcd member on the same node. This avoids overwhelming a specific member with backup requests.
+
+Since a Backup pod exist on every master node, and a `hostPath` volume is being used to store the backups, it is guaranteed to have available backups on all master nodes, which is required for disaster recovery scenarios.
+
+_**Cons**_
+
+No issue has been encountered during testing this approach.
+
+### Open Questions
+
+- What values could be used for default configurations, considering all OCP variants.
+- What is the minimum disk size needed in order to operate this feature without running out of disk space ?
+- How to handle an empty `schedule` ? 
+- How to mitigate losing backups upon control-plane scaling using `CPMSO` ?
```

Comment:
I don't see why should we should worry about this - if the backups are interchangable any other backup would do. If they are not you won't be able to restore this node anyway

Reply:
If something goes drastically wrong during a CPMS roll out and you end up in a position where you've lost all your backups, you'll probably be pretty annoyed.

If the backup pods are running as daemonsets and running always, perhaps they should check on startup for the presence of any backups, and, produce a backup shortly after starting if nothing else is present?

That would hopefully mean that by the time the CPMS has finished rolling the second node, the first new node will have at least one backup, so there is at least some backup somewhere

Reply:
> If the backup pods are running as daemonsets and running always, perhaps they should check on startup for the presence of any backups, and, produce a backup shortly after starting if nothing else is present?

I think producing a backup from a fresh node, should take place only in either of two cases are valid 

- the etcd member on the new node is no longer `learner` (i.e. it is a voting member of the etcd cluster)
- the backup pod gets a backup copy from other node

I think the latter is  unnecessary complicated, and the first is the most straightforward solution 

That being said, I think this case could be handled best using backups stored on `External Targets` (e.g. S3 bucket)

Reply:
raised this to track https://issues.redhat.com/browse/ETCD-694

feel free to comment to shape the requirements

Reply:
Seems reasonable to wait for the member to be voting, +1 to that

And you're correct, the external backup is the ultimate solution, but, if we did a one off backup straight after the etcd member became voting, that would make this feature, in combination with CPMS, much more reliable

Reply:
@deads2k I need your opinion here, what to do with a new member joining the cluster ? 

see above